frontend/src/composables/use-focus-on-show.ts

```diff
@@ -48,7 +50,9 @@ export const useFocusOnShow = ({
       // We already do that in a more flexible, adaptive way in our Dialog composables.
       initialFocus: false,
       // if set to true, focus-trap prevents the default for the keyboard event, and we cannot handle it in our composables.
-      escapeDeactivates: false, // Even though we pass `initialFocus` as `false` above, `focus-trap` still
+      escapeDeactivates: false,
+      clickOutsideDeactivates: () => hideOnClickOutsideRef.value,
+      // Even though we pass `initialFocus` as `false` above, `focus-trap` still
```

Comment:
Should we eventually re-evaluate whether using focus-trap in this way makes sense? Can we either remove it entirely somehow or else rely on it exclusively for focus management of dialogs?

Reply:
@obulat any advice here? Is there anything we should reconsider here and if so, can it be turned into an issue? If not, I'd just like confirmation for my own understanding, though documenting it in the code would also be helpful and could be its own issue.